src/macad_gym/envs/follow_leading_vehicle.py

```diff
@@ -0,0 +1,139 @@
+"""
+Author: Morphlng
+Date: 2023-01-28 18:33:26
+LastEditTime: 2023-01-28 18:33:28
+LastEditors: Morphlng
+Description: Example of creating a custom environment, also a demonstration of how to use the manual_control.
+FilePath: \src\macad_gym\envs\follow_leading_vehicle.py
+"""
+
+import random
+from macad_gym.envs import MultiCarlaEnv
+
+"""
+    This is a scenario extracted from Carla/scenario_runner (https://github.com/carla-simulator/scenario_runner/blob/master/srunner/scenarios/follow_leading_vehicle.py).
+
+    The configuration below contains everything you needed to customize your own scenario in Macad-Gym.
+"""
+configs = {
+    "scenarios": {
+        "map": "Town01",
+        "actors": {
+            "car1": {
+                "start": [107, 133, 0.5],
+                "end": [300, 133, 0.5],
+            },
+            "car2": {
+                "start": [115, 133, 0.5],
+                "end": [310, 133, 0.5],
+            }
+        },
+        "num_vehicles": 0,
+        "num_pedestrians": 0,
+        "weather_distribution": [0],
+        "max_steps": 500
+    },
+    "env": {
+        "server_map": "/Game/Carla/Maps/Town01",
+        "render": False,
+        "render_x_res": 800,    # For both Carla-Server and Manual-Control
+        "render_y_res": 600,
+        "x_res": 84,            # Used for camera sensor view size
+        "y_res": 84,
+        "framestack": 1,
+        "discrete_actions": True,
+        "squash_action_logits": False,
+        "verbose": False,
+        "use_depth_camera": False,
+        "send_measurements": False,
+        "enable_planner": True,
+        "spectator_loc": [70, -125, 9],
+        "sync_server": True,
+        "fixed_delta_seconds": 0.05,
+    },
+    "actors": {
+        "car1": {
+            "type": "vehicle_4W",
+            "enable_planner": True,
+            "convert_images_to_video": False,
+            "early_terminate_on_collision": True,
+            "reward_function": "corl2017",
+            "scenarios": "FOLLOWLEADING_TOWN1_CAR1",
+            "manual_control": True,     # manual_control and auto_control
+            "auto_control": True,       # can be turned on at the same time
```

Comment:
To avoid confusion, please update the comment to say: "When True, starts the actor in auto-pilot and allows manual control take-over on pressing Key `p` on the PyGame window if `manual_control` is also True"